Source/WebKit/Shared/cf/CoreIPCSecTrust.mm

```diff
@@ -0,0 +1,343 @@
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#if HAVE(WK_SECURE_CODING_SECTRUST) && USE(CF)
+
+#import "config.h"
+#import "CoreIPCSecTrust.h"
+
+#import <wtf/Compiler.h>
+#import <wtf/cocoa/TypeCastsCocoa.h>
+
+namespace WebKit {
+
+
+CoreIPCSecTrust::CoreIPCSecTrust(SecTrustRef trust)
+{
+    CFErrorRef error;
+
+    if (!trust)
+        return;
+
+    auto dict = adoptNS(bridge_cast(static_cast<CFDictionaryRef>(SecTrustCopyPropertyListRepresentation(trust, &error))));
+
+    if (error || ![dict isKindOfClass:NSDictionary.class])
+        return;
+
+    CoreIPCSecTrustData secTrustData { };
+
+    NSDate *verifyDate = [dict objectForKey:@"verifyDate"];
+    if ([verifyDate isKindOfClass:NSDate.class]) {
+        auto d = CoreIPCDate(verifyDate);
+        secTrustData.verifyDate = WTFMove(d);
+    }
+
+ALLOW_DEPRECATED_DECLARATIONS_BEGIN
+    NSNumber *result = [dict objectForKey:@"result"];
+    if ([result isKindOfClass:NSNumber.class]) {
+        switch ([result unsignedCharValue]) {
+        case kSecTrustResultInvalid:
+            secTrustData.result = CoreIPCSecTrustResult::Invalid;
+            break;
+        case kSecTrustResultProceed:
+            secTrustData.result = CoreIPCSecTrustResult::Proceed;
+            break;
+        case kSecTrustResultConfirm:
+            secTrustData.result = CoreIPCSecTrustResult::Confirm;
+            break;
+        case kSecTrustResultDeny:
+            secTrustData.result = CoreIPCSecTrustResult::Deny;
+            break;
+        case kSecTrustResultUnspecified:
+            secTrustData.result = CoreIPCSecTrustResult::Unspecified;
+            break;
+        case kSecTrustResultRecoverableTrustFailure:
+            secTrustData.result = CoreIPCSecTrustResult::RecoverableTrustFailure;
+            break;
+        case kSecTrustResultFatalTrustFailure:
+            secTrustData.result = CoreIPCSecTrustResult::FatalTrustFailure;
+            break;
+        case kSecTrustResultOtherError:
+            secTrustData.result = CoreIPCSecTrustResult::OtherError;
+            break;
+        }
+    }
+ALLOW_DEPRECATED_DECLARATIONS_END
+
+    NSNumber *anchorsOnly = [dict objectForKey:@"anchorsOnly"];
+    if ([anchorsOnly isKindOfClass:NSNumber.class])
+        secTrustData.anchorsOnly = [anchorsOnly boolValue];
+
+    NSArray *certificates = [dict objectForKey:@"certificates"];
+    if ([certificates isKindOfClass:NSArray.class]) {
+        Vector<CoreIPCData> vector;
+        vector.reserveCapacity(certificates.count);
+        for (NSData* item in certificates) {
+            if (![item isKindOfClass:NSData.class]) {
+                ASSERT_NOT_REACHED();
+                return;
+            }
+            auto c = CoreIPCData(item);
+            vector.append(WTFMove(c));
```

Comment:
Can we do `vector.append(CoreIPCData { item });`

Reply:
For all of these you get a compile error saying that it cannot do copy elision